### PR TITLE
5357 metadata

### PR DESCRIPTION
**Pull request by pgarrison**
Save metadata from twitter, facebook for post-hoc analysis.

### CODE REVIEW

File: lib/fetching/content-services/facebook-content-service.js [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
+out by the _lastReportDate.
 
   - Service does not use the "since" param because of the different responses for the different object types.
 
   - Service does not use paging because of usage limits.
 
-  - This service will miss any posts beyond 10 that have been made since the last fetch. e.g If 12 posts were made since
-  the last fetch the 1rst and 2nd(oldest) posts will be missed.
+  - This service will miss any posts beyond 10 that have been made since the last fetch.
+  e.g If 12 posts were made since the last fetch the 1rst and 2nd(oldest) posts will be missed.
```
Comment (pgarrison) [inline lib/fetching/content-services/facebook-content-service.js]: Tiniest nit: we should indent all of these the same way.

File: lib/fetching/content-services/facebook-content-service.js [hunk 2]
```diff
@@ -51,8 +58,17 @@ var FacebookContentService = function(options) {
...
   this._url = options.url;
+  /*
+  Stuff that can be collected
+  likes.total_count,
+  reactions.total_count,
+  place.id,
+  place.location.latitude,
+  place.location.longitude,
+  comments.total_count,
+  */
```
Comment (pgarrison) [inline lib/fetching/content-services/facebook-content-service.js]: We should get rid of this, and the Twitter one

File: lib/fetching/content-services/facebook-content-service.js [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
 
-  - This service will miss comments on any posts more than 10 posts old. Comments are attached the posts, so no comments
-  on any posts older than the 10th oldest one will be retrieved.
+  - This service will miss comments on any posts more than 10 posts old.
+    Comments are attached the posts, so no comments on any posts older than the 10th oldest one
+    will be retrieved.
 
   - It will also miss any comments beyond 10 that have been made since the last fetch.
 */
+'use strict';
```
Comment (pgarrison) [inline lib/fetching/content-services/facebook-content-service.js]: I think if we're going to strict mode here we need to fix the use of `this` that eslint notices.

File: lib/fetching/content-services/twitter-content-service.js [hunk 4]
```diff
@@ -80,11 +80,24 @@ TwitterContentService.prototype._getStream = function() {
...
+    longitude: data.coordinates ? data.coordinates[1] : 0,
+    retweetCount: data.retweet_count ? data.retweet_count : 0,
+    favouriteCount: data.favorite_count ? data.favorite_count : 0
+  };
   return {
     authoredAt: new Date(data.created_at),
     fetchedAt: new Date(),
     content: data.text,
     author: author,
+    metadata: metadata,
```
Comment (pgarrison) [inline lib/fetching/content-services/twitter-content-service.js]: Codeclimate caught this extra comma here.

Comment (karthiksbhat) [inline lib/fetching/content-services/twitter-content-service.js]: Codeclimate caught the one in the facebook content service, and I've fixed that. :)

**Comment by pgarrison** [comment]
Let's check in e.g. `lib.fetching.content-services.facebook-content-service.test.js` that reports get emitted with metadata.

**Pull request merged**